src/AsmResolver/FieldReader.cs

```diff
@@ -0,0 +1,26 @@
+﻿using System.Reflection;
+
+namespace AsmResolver
+{
+    public class FieldReader
```

Comment:
Define as `static`. Also this is not really used outside of `AsmResolver.DotNet`. Not sure if placing it in the `AsmResolver` project is correct. If moved to `AsmResolver.DotNet`, mark it `internal` instead of `public` as well.